src/Item_Line.php

```diff
@@ -0,0 +1,576 @@
+            'WHERE'  => [
+                'lines_id' => $ID,
+            ]
+        ]);
+
+        $simcards = $DB->request([
+            'FROM'   => Item_DeviceSimcard::getTable(),
+            'WHERE'  => [
+                'lines_id' => $ID
+            ]
+        ]);
+
+        Session::initNavigateListItems(
+            self::getType(),
+            //TRANS : %1$s is the itemtype name,
+            //        %2$s is the name of the item (used for headings of a list)
+            sprintf(
+                __('%1$s = %2$s'),
+                $line->getTypeName(1),
+                $line->getName()
+            )
+        );
+
+        $items = iterator_to_array($items);
+        $simcards = iterator_to_array($simcards);
+
+        if (!count($simcards)) {
+            echo "<table class='tab_cadre_fixe'><tr><th>" . __('No simcard found') . "</th></tr>";
+            echo "</table>";
+        } else {
+            echo "<table class='tab_cadre_fixehov'>";
+            $header = "<tr>";
+            $header .= "<th>" . Item_DeviceSimcard::getTypeName(1) . "</th>";
+            $header .= "</tr>";
+
+            echo $header;
+            foreach ($simcards as $row) {
+                $item = new Item_DeviceSimcard();
+                $item->getFromDB($row['id']);
+                echo "<tr class='tab_bg_1'>";
+                echo "<td>" . $item->getLink() . "</td>";
+                echo "</tr>";
+            }
+            echo $header;
+            echo "</table>";
+        }
+
+        if ($line->canAddItem('itemtype')) {
+            echo "<div class='firstbloc'>";
+            Html::showSimpleForm(
+                self::getFormURL(),
+                '_add_fromitem',
+                __('Add new item to this line...'),
+                [
+                    'line'   => $ID
+                ]
+            );
+            echo "</div>";
+        }
+
+        if (!count($items)) {
+            echo "<table class='tab_cadre_fixe'><tr><th>" . __('No item found') . "</th></tr>";
+            echo "</table>";
+        } else {
+            if ($canedit) {
+                Html::openMassiveActionsForm('mass' . __CLASS__ . $rand);
+                $massiveactionparams = [
+                    'num_displayed'   => min($_SESSION['glpilist_limit'], count($items)),
+                    'container'       => 'mass' . __CLASS__ . $rand
+                ];
+                Html::showMassiveActions($massiveactionparams);
+            }
+
+            echo "<table class='tab_cadre_fixehov'>";
+            $header = "<tr>";
+            if ($canedit) {
+                $header .= "<th width='10'>";
+                $header .= Html::getCheckAllAsCheckbox('mass' . __CLASS__ . $rand);
+                $header .= "</th>";
+            }
+            $header .= "<th>" . _n('Item', 'Items', 1) . "</th>";
+            $header .= "</tr>";
+
+            echo $header;
+            foreach ($items as $row) {
+                $item = new $row['itemtype']();
```

Comment:
Relation tables may sometimes contains items added by plugins that are no longer active.
```suggestion
                if (!is_a($row['itemtype'], CommonDBTM::class, true)) {
                    continue;
                }
                $item = new $row['itemtype']();
```